core/src/components/menu/menu.tsx

```diff
@@ -148,6 +148,17 @@ export class Menu implements ComponentInterface, MenuI {
 
     const el = this.el;
     const parent = el.parentNode as any;
+    if (this.contentId === undefined) {
+      console.warn(`[DEPRECATED][ion-menu] Using the [main] attribute is deprecated, please use the "contentId" property instead:
+BEFORE:
+  <ion-menu>...</ion-menu>
+  <div main>...</div>
+
+AFTER:
+  <ion-menu contentId="my-content"></ion-menu>
```

Comment:
```suggestion
  <ion-menu content-id="my-content"></ion-menu>
```

Should this be changed for vanilla or only angular?

Reply:
good question

Reply:
I am guessing since react and vue aren't released yet we should write it like angular so this is good